pkg/minikube/download/image.go

```diff
@@ -224,6 +226,50 @@ func ImageToCache(img string) error {
 	}
 }
 
+func DownloadImageFromGithubToCache(img string, imgVersion string) (string, error) {
+	f := imagePathInCache(img)
+	fileLock := f + ".lock"
+
+	kicbaseArch := runtime.GOARCH
+	if kicbaseArch == "arm" {
+		kicbaseArch = "armv7"
+	}
+
+	releaser, err := lockDownload(fileLock)
+	if err != nil {
+		return "", err
+	}
+	if releaser != nil {
+		defer releaser.Release()
+	}
+	downloadURL := fmt.Sprintf("https://github.com/kubernetes/minikube/releases/download/%s/%s-%s-%s.tar",
+		version.GetVersion(),
+		img, imgVersion, kicbaseArch)
+	// if we reach here, that means the user cannot have access to any docker registry
+	// this means the user is very likely to have a network issue
+	// downloading from github via http is the last resort, and we should remind the user
+	// that he should at least get access to github
+	// print essential warnings
+	out.WarningT("minikube cannot pull kicbase image from any docker registry, and is trying to download kicbase tarball from github release page via HTTP.")
+	out.Ln("The url to the tarball is %s", downloadURL)
+	out.WarningT("It's very likely that you have an internet issue. Please ensure that you can access the internet at least via HTTP, directly or with proxy. Currently your proxy configure is:")
+	envs := []string{"HTTP_PROXY", "HTTPS_PROXY", "http_proxy", "https_proxy", "ALL_PROXY", "NO_PROXY"}
+	for _, env := range envs {
+		if v := os.Getenv(env); v != "" {
+			out.Infof("{{.env}}={{.value}}", out.V{"env": env, "value": v})
+		}
+	}
+	out.Ln("")
```

Comment:
Can you move this to before we call the function in `cache.go`, we probably want to have it where if the user sets their country to China in the flag we call this directly and don't want to output this message for them.